src/towerside/sequence.cpp

```diff
@@ -0,0 +1,176 @@
+#include "sequence.hpp"
+#include "common/config.hpp"
+#include "config.hpp"
+
+namespace sequence {
+
+void set_state(State &state, ActuatorMessage &current_cmd, unsigned long &start_time) {
```

Comment:
Output parameters should always be passed by pointer not reference, so its clear at the callsite that its an output parameter. That being said I think you should just have the sequence state machine live in this file, so that `main()` just does the simple automatic/manual check and calls into `sequence.cpp` if we're in automatic mode to do the rest of the logic.